schemas/oic.rule-rw-schema.json

```diff
@@ -0,0 +1,40 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "description" : "Copyright (c) 2018 Open Connectivity Foundation, Inc. All rights reserved.",
+    "id": "http://openconnectivityfoundation.github.io/core/schemas/oic.rule-rw-schema.json#",
+    "title": "Rule Resource Type",
+    "definitions": {
+      "rule-properties": {
+        "properties": {
+          "rule": {
+            "type": "string",
+            "format": "ocf-rule-string",
+            "maxLength": 8192,
+            "readOnly": false,
+            "description": "The rule itself, expressed as a string in the ABNF format specified"
+          },
+          "ruleresult": {
+            "type": "boolean",
+            "readOnly": false,
+            "description": "The boolean result of evaluating the rule with the current inputs"
+		      },
+          "ruleenable": {
+            "type": "boolean",
+            "readOnly": false,
+            "description": "If true, the rule result is updated whenever any rule input is updated"
+		      },
+          "actionenable": {
+            "type": "boolean",
+            "readOnly": false,
+            "description": "If true, actions are triggered on ruleenable value transition from false to true"
+		      }
+        }
+      }
+    },
+    "type": "object",
+    "allOf": [
+      {"$ref": "#/definitions/rule-properties"}
+    ],
+    "required": ["rule","ruleresult","ruleenable","actionenable"],
```

Comment:
I think we need two schemas:  One for RETRIEVE that has all four Properties required and one for UPDATE to allow you to modify one Property at a time.  In other words, you should be able to send an UPDATE request with just one Property and not include the other three, right?